lib/room.cpp

```diff
@@ -402,6 +408,161 @@ class Room::Private {
+                        if(data["one_time_keys"][user][device].toObject().isEmpty()) {
+                            qWarning() << "No one time key for" << user << device;
+                            continue;
+                        }
+                        const auto keyId = data["one_time_keys"][user][device].toObject().keys()[0];
+                        const auto oneTimeKey = data["one_time_keys"][user][device][keyId]["key"].toString();
+                        const auto signature = data["one_time_keys"][user][device][keyId]["signatures"][user][QStringLiteral("ed25519:") + device].toString().toLatin1();
+                        auto signedData = data["one_time_keys"][user][device][keyId].toObject();
+                        signedData.remove("unsigned");
+                        signedData.remove("signatures");
+                        auto signatureMatch = QOlmUtility().ed25519Verify(connection->edKeyForUserDevice(user, device).toLatin1(), QJsonDocument(signedData).toJson(QJsonDocument::Compact), signature);
+                        if (!signatureMatch) {
+                            qCWarning(E2EE) << "Failed to verify one-time-key signature for" << user << device << ". Skipping this device.";
+                            continue;
+                        } else {
+                        }
+                        connection->createOlmSession(recipientCurveKey, oneTimeKey);
+                    }
+                    usersToDevicesToEvents[user][device] = payloadForUserDevice(user, device, sessionId, sessionKey);
+                }
+            }
+            if (!usersToDevicesToEvents.empty()) {
+                connection->sendToDevices("m.room.encrypted", usersToDevicesToEvents);
+                QHash<QString, QList<std::pair<QString, QString>>> receivedDevices;
+                for (const auto& user : devices.keys()) {
+                    for (const auto& device : devices[user]) {
+                        receivedDevices[user] += {device, q->connection()->curveKeyForUserDevice(user, device) };
+                    }
+                }
+                connection->database()->setDevicesReceivedKey(q->id(), receivedDevices, sessionId, index);
+            }
+        });
+    }
+
+    void sendMegolmSession(const QHash<QString, QStringList>& devices) {
+        // Save the session to this device
+        const auto sessionId = currentOutboundMegolmSession->sessionId();
+        const auto _sessionKey = currentOutboundMegolmSession->sessionKey();
+        if(!_sessionKey) {
+            qCWarning(E2EE) << "Error loading session key";
+            return;
+        }
+        const auto sessionKey = *_sessionKey;
+        const auto senderKey = q->connection()->olmAccount()->identityKeys().curve25519;
+
+        // Send the session to other people
+        sendRoomKeyToDevices(sessionId, sessionKey, devices, currentOutboundMegolmSession->sessionMessageIndex());
```

Comment:
Just a small brush-up.
```suggestion
        const auto sessionKey = currentOutboundMegolmSession->sessionKey();
        if(!sessionKey) {
            qCWarning(E2EE) << "Error loading session key";
            return;
        }
        const auto senderKey = q->connection()->olmAccount()->identityKeys().curve25519;

        // Send the session to other people
        sendRoomKeyToDevices(sessionId, *sessionKey, devices, currentOutboundMegolmSession->sessionMessageIndex());
```